### ISSUE TITLE
Fix warning in SAT_Clause.hpp

### ISSUE THREAD

**Issue by sfrehse** [agra-uni-bremen/metaSMT#13]
/metaSMT/build/root/include/metaSMT/backend/SAT_Clause.hpp:71:47: warning: narrowing conversion of 'metaSMT::impl::new_var_id()' from 'unsigned int' to 'int' inside { } [-Wnarrowing]

`````` C++
result_type operator() (logic::tag::or_tag const& tag, result_type lhs, result_type rhs )
{
        result_type out = { impl:: new_var_id() }; // this line states the problem. 
        clause3( lhs, rhs,-out);
        clause2(-rhs, out);
        clause2(-lhs, out);
        return out;
}```

``````

**Comment by sfrehse** [comment]
`impl::new_var_id` returns an `unsigned` but result_type expected an `int`. Is that safe to cast?

**Comment by finnhaedicke** [comment]
for the first 2 billion variables it should be save.

**Comment by sfrehse** [comment]
Okay, I'm going to fix that. Finally we'll get approximately hundreds of thousands fewer warnings :+1:.

**Comment by finnhaedicke** [comment]
fixed in 22f49ec900482d81468454d5320d4f2ad7fd4666
